src/buildercore/lifecycle.py

```diff
@@ -111,17 +116,30 @@ def start(stackname):
         LOG.info("RDS nodes to be started: %s", rds_to_be_started)
         [_rds_connection(stackname).start_db_instance(DBInstanceIdentifier=n) for n in rds_to_be_started]
 
-    if ec2_to_be_started:
-        _wait_ec2_all_in_state(stackname, 'running', ec2_to_be_started)
+    if not ec2_to_be_started:
+        LOG.info("EC2 nodes are all running")
+
+    def _wait_for_ec2_steady_state():
+        _wait_ec2_all_in_state(stackname, 'running', ec2_to_be_checked)
         call_while(
             lambda: _some_node_is_not_ready(stackname),
             interval=2,
-            timeout=120,
+            timeout=config.BUILDER_TIMEOUT,
             update_msg="waiting for nodes to complete boot",
-            done_msg="all nodes have public ips, are reachable via SSH and have completed boot"
+            done_msg="all nodes have public ips, are reachable via SSH and have completed boot",
+            exception_class=EC2Timeout
         )
-    else:
-        LOG.info("EC2 nodes are all running")
+    try:
+        _wait_for_ec2_steady_state()
+    except EC2Timeout:
+        # in case of botched boot and/or inability to
+        # access through SSH, try once to stop and
+        # start the instances again
+        LOG.info("Retrying boot on %s", ec2_to_be_checked)
+        _ec2_connection(stackname).instances.filter(InstanceIds=ec2_to_be_checked).stop()
+        _wait_ec2_all_in_state(stackname, 'stopped', ec2_to_be_checked)
+        _ec2_connection(stackname).instances.filter(InstanceIds=ec2_to_be_checked).start()
```

Comment:
we're not doing any caching of the connection and I don't think the boto library would be re-using connections either. I wonder if there is any overhead or extra noise because of the new connections. And if it's even worth a second thought..